### DISCUSSION TITLE
After running a C# Script the Button for Save back to model dis-appears

### DISCUSSION THREAD

**Discussion by MikeCarlo** [TabularEditor/TabularEditor3#1068] [Ideas]
# Issue

One of the first C# scripts I run is to create a group of sum measures, Add them to a display folder, set column format, then hide the numerical columns.  While on this screen I have the following buttons on the menu:

![image](https://user-images.githubusercontent.com/25160592/136683771-33978aa3-5fa5-4e59-9960-83b819c9808e.png)

The third icon is the save button for the C# script.  

When I want to save my changes back to the data model, I have to click on another window such as Expression Editor. Then the icons at the top menu will change to the following:

![image](https://user-images.githubusercontent.com/25160592/136683763-394af8bf-c28f-4c24-8221-45b829f67a7a.png)

I feel this is confusing.  Often I have clicked "Save" while I was in the C# window, Thinking I was saving my changes back to the model.  when in fact I was just saving the C# script file. 

I would prefer when I navigate to the Script window to have an Added icon for saving the script.  Please don't remove the save changes back to model.

**Comment by otykier** [comment]
For consistency with other applications (like Visual Studio), the CTRL+S hotkey, in my opinion, should always be linked to whichever document is currently in focus. But I agree that we need a "Save model" action which stays on the toolbar, regardless of which document currently has focus. In addition, I think we should come up with a new hotkey for that: ALT+S

So by default, the hotkeys would be:

- **CTRL+S**: Saves the current document (or the model, if no document has focus)
- **ALT+S**: Saves the model
- **CTLR+SHIFT+S**: Save all

Thoughts?

**Comment by otykier** [comment]
Also (from https://github.com/TabularEditor/TabularEditor/issues/939) we should make sure to display "ALT+S" in the information bar, when reminding users that changes have been made which have not yet been synchronized.
